yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -569,7 +570,7 @@ private[spark] class Client(
       logApplicationReport: Boolean = true): (YarnApplicationState, FinalApplicationStatus) = {
     val interval = sparkConf.getLong("spark.yarn.report.interval", 1000)
     var lastState: YarnApplicationState = null
-    while (true) {
+    while (!stopMonitor) {
```

Comment:
Oops, I mean you would have to catch `InterruptedException` with a `try` block around this loop, and log and/or ignore it.